x/distribution/types/store.go

```diff
@@ -0,0 +1,35 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// historical rewards for a validator
+// TODO add reference counter, ref https://github.com/cosmos/cosmos-sdk/pull/3099#discussion_r245747051
+// height is implicit within the store key
+type ValidatorHistoricalRewards = sdk.DecCoins
+
+// current rewards for a validator
+type ValidatorCurrentRewards struct {
+	Rewards sdk.DecCoins `json:"rewards"`
+	Period  uint64       `json:"period"`
+}
+
+// accumulated commission for a validator
+type ValidatorAccumulatedCommission = sdk.DecCoins
+
+// starting info for a delegator reward period
+type DelegatorStartingInfo struct {
```

Comment:
What exactly is this struct and why is it needed? Is `Stake` the amount of initially bonded tokens? Also, why track the previous period specifically?

Reply:
Added the above as comments in the file.